app/controllers/oauth_controller.rb

```diff
@@ -54,6 +54,12 @@ def callback
       begin
```

Comment:
Hey @ktp242!

Two questions....

1.  Why do we delete these keys _a la carte_ here? Can we just call `reset_session`?
2.  Sorry I didn't see this earlier but....Is there a reason why we store the OAuth response in `self.class.token`?
Seems like the class level is a place to more appropriately store state applicable to all instances of `OAuthController`? Maybe if we store it in a local variable it would read easier?

All together this diff would look like.

```diff
diff --git a/app/controllers/oauth_controller.rb b/app/controllers/oauth_controller.rb
index b1aa63f..e7c7df4 100644
--- a/app/controllers/oauth_controller.rb
+++ b/app/controllers/oauth_controller.rb
@@ -53,16 +53,12 @@ class OauthController < ApplicationController
       # Catch the error and proceed to redirenct to '/' if we fail to get the access token
       begin
         # Get the access token and initialize it with ACCESS_TOKEN class
-        self.class.token = OAUTH_CLIENT.auth_code.get_token(params[:code], :redirect_uri => ENV['OAUTH_CALLBACK_URL'])
+        oauth_response = OAUTH_CLIENT.auth_code.get_token(params[:code], :redirect_uri => ENV['OAUTH_CALLBACK_URL'])
+        reset_session
 
-        # Clear the old session, if we happened to have any
-        session.delete(:access_token)
-        session.delete(:refresh_token)
-        session.delete(:access_token_expires_at)
-
-        session[:access_token] = self.class.token.token
-        session[:refresh_token] = self.class.token.refresh_token
-        session[:access_token_expires_at] = self.class.token.expires_at
+        session[:access_token] = oauth_response.token
+        session[:refresh_token] = oauth_response.refresh_token
+        session[:access_token_expires_at] = oauth_response.expires_at
 
         # Now we can put "Authorization: bearer #{session[:access_token]}" in the header when making an HTTP request
         # Or, we can use self.class.token.get self.class.token.post to make requests

```

Reply:
1. Updated the method to use iteration to cleat the old session! Thanks for @nodanaonlyzuul 's suggestion. We can't clear the whole session because one of the session is used to stored the URL we want to redirect the user to.

2. For now `self.class.token` is assigned only for the use of refresh token method. But I noticed some flaws on my refresh token method. It didn't seem working 100% of the time, so I will have another PR to fix it. And the fix won't need `self.class.token` anymore, so it will be gone eventually.